python/bindings/openravepy_int.cpp

```diff
@@ -124,40 +124,59 @@ object toPyObject(const rapidjson::Value& value)
     }
 }
 
-#define FILL_RAPIDJSON_FROMARRAY_1D(pyarrayvalues, T, rapidjsonsetfn) {  \
-        const T *vdata = reinterpret_cast<T*>(PyArray_DATA(pyarrayvalues)); \
-        for (int i = 0; i < dims[0]; i++) { \
-            rapidjson::Value elementValue; \
-            elementValue.rapidjsonsetfn(vdata[i]); \
-            value.PushBack(elementValue, allocator); \
-        } \
-} \
-
-#define FILL_RAPIDJSON_FROMARRAY_2D(pyarrayvalues, T, rapidjsonsetfn) {                            \
-        const T *vdata = reinterpret_cast<T*>(PyArray_DATA(pyarrayvalues)); \
-        for (int i = 0; i < dims[0]; i++) { \
-            rapidjson::Value colvalues(rapidjson::kArrayType); \
-            for (int j = 0; j < dims[1]; j++) { \
-                rapidjson::Value elementValue; \
-                elementValue.rapidjsonsetfn(vdata[i*dims[0]+j]); \
```

Comment:
I just noticed the index `i*dims[0]+j` is not right in the first place, is it? @rdiankov

Reply:
yup, it was wrong; should be i*dims[1]+j as expected